File: c_plus_plus/ferris_wheel.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main(){
    int n,a,l,r,ans;
    cin >> n >> a;
    r = n - 1,l = 0,ans = 0;

    int* arr = new int[n];
    for(int i = 0;i < n;i++){
        cin >> arr[i];
    }

    sort(arr,arr + n);

    while(l <= r){
        if(arr[l] + arr[r] <= a){
            l += 1;
        }
        r -= 1;
        ans += 1;
    }

    cout << ans;
}

File: baitap_project/on_thi/Ferris_Wheel.py
a,b = list(map(int,input().split()))
arr = list(map(int,input().split()))
arr.sort()
l = 0
r = a - 1
ans = 0
while l <= r:
    if arr[l] + arr[r] <= b:
        l += 1
    r -= 1
    ans += 1

print(ans)